geps/gep-1897.md

```diff
@@ -25,11 +31,18 @@ Gateway to backend**, such as TLS versions and cipher suites. (Use case #5 in [G
 
 ## Longer Term Goals
 
-These are worthy goals, but may need a different GEP for proper attention.
+These are worthy goals, but may need a different GEP for proper attention.  This GEP is concerned entirely with the
+controlplane, i.e. the hop between gateway and backend.
 
-1. TCPRoute use cases (completed by GA)
-2. Mutual TLS use cases
-3. Service mesh use cases
+1. [TCPRoute](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.TCPRoute) and
+[GRPCRoute](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.GRPCRoute) use cases
+are not addressed here, because at this point in time these two route types are not graduated to beta.
+2. Mutual TLS (mTLS) use cases are intentionally out of scope for this GEP because this GEP does not address the case
```

Comment:
Sorry to keep hammering on this point, but I am not sure this makes sense and it was my only pushback on this GEP so I think its important to get it right.

**Mutual TLS being out of scope is not because of mesh.** 

>  It is about the  case where an application developer needs to explicitly express that they expect TLS when there is no automatic, implicit configuration of that.

Its perfectly valid for an implementation to expect mTLS explicitly. mTLS is not even really an independent thing, its just TLS with a field set - its a name for a subset of TLS.

Mutual TLS is not addressed by the GEP because _the API fundamentally cannot support mutual authentication_ because of its design of being backend-attached.

tl;dr

* Not supporting "mesh transport" is an explicit design goal
* Not supporting "mutual TLS" is a down-scoping of the GEP to make sure we can get something merged. However, the design is not future proofed, so despite this being a longer term goal it cannot be extended
* Mutual TLS != Mesh Transport

Reply:
My goal was to highlight mesh as an example of the "case where connections to TLS are implicitly configured on the behalf of the user".  I will try to add that the "API cannot support mutual authentication because its design is backend-attached".